codec.go

```diff
@@ -113,3 +119,33 @@ func (c *Codec) HardwareConfigs(dt HardwareDeviceType) (configs []CodecHardwareC
 	}
 	return
 }
+
+func (c *Codec) HasHardwareConfigMethodFlag(chcmf CodecHardwareConfigMethodFlag) bool {
+	var i C.int
+	for {
+		config := C.avcodec_get_hw_config(c.c, i)
+		if config == nil {
+			break
+		}
+
+		if (CodecHardwareConfig{c: config}.MethodFlags().Has(chcmf)) {
+			return true
+		}
+
+		i++
+	}
+	return false
+}
+
+func IterateCodecs(processor CodecProcessor) {
```

Comment:
I'd rather have a global `Codecs() []*Codec` method returning all available codecs

Reply:
ok but would u think its not better to have also av_codec_iterate in the lib

Reply:
I mean at end it would be the Same :)